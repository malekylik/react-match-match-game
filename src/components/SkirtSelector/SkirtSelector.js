import React from 'react';
import PropTypes from 'prop-types';

const SkirtSelector = ({ id, imageSrc: src, width, height, alt, onClick, defaultChecked = false }) => (
    <p>
        <label htmlFor={id}>
            <img width={width} height={height} src={src} alt={alt} />
        </label>
        <input id={id} type="radio" name="skirt" onClick={onClick} defaultChecked={defaultChecked} />
    </p>
);

SkirtSelector.propTypes = {
    id: PropTypes.string.isRequired,
    imageSrc: PropTypes.string.isRequired,
    width: PropTypes.string.isRequired,
    height: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    defaultChecked: PropTypes.bool,
};

export default SkirtSelector;
