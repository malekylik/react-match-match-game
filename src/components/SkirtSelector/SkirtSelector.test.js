import React from 'react';

import { shallow } from 'enzyme';

import SkirtSelector from './SkirtSelector';

describe('SkirtSelector', () => {
    it('Should match snapshot', () => {
        expect(shallow(
            <SkirtSelector
                key="skirt-1"
                id="skirt1"
                imageSrc="test"
                width="45"
                height="70"
                alt="skirt1"
                onClick={jest.fn()}
                defaultChecked={true} />)).toMatchSnapshot();
    });
});
