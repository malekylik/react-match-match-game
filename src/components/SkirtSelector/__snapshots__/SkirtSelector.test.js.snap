// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkirtSelector Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkirtSelector
    alt="skirt1"
    defaultChecked={true}
    height="70"
    id="skirt1"
    imageSrc="test"
    onClick={[MockFunction]}
    width="45"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="skirt1"
        >
          <img
            alt="skirt1"
            height="70"
            src="test"
            width="45"
          />
        </label>,
        <input
          defaultChecked={true}
          id="skirt1"
          name="skirt"
          onClick={[MockFunction]}
          type="radio"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="skirt1"
            height="70"
            src="test"
            width="45"
          />,
          "htmlFor": "skirt1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "skirt1",
            "height": "70",
            "src": "test",
            "width": "45",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "defaultChecked": true,
          "id": "skirt1",
          "name": "skirt",
          "onClick": [MockFunction],
          "type": "radio",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="skirt1"
          >
            <img
              alt="skirt1"
              height="70"
              src="test"
              width="45"
            />
          </label>,
          <input
            defaultChecked={true}
            id="skirt1"
            name="skirt"
            onClick={[MockFunction]}
            type="radio"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="skirt1"
              height="70"
              src="test"
              width="45"
            />,
            "htmlFor": "skirt1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "skirt1",
              "height": "70",
              "src": "test",
              "width": "45",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultChecked": true,
            "id": "skirt1",
            "name": "skirt",
            "onClick": [MockFunction],
            "type": "radio",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
